setup.py

```diff
@@ -51,6 +51,7 @@ def get_version():
         'setuptools',
         'numpy',
         'iminuit',
+        'matplotlib'
```

Comment:
It would be better if matplotlib is optional.

In the packages I know, this is done by putting the `matplotlib` import into the function or method, i.e. have it delayed and only happen when it's needed.
```
def compute_something():
    # can be used without matplotlib

def plot_something():
    import matplotlib
```
Then for the tests the ones requiring optional dependencies are marked like this:
http://docs.astropy.org/en/latest/development/testguide.html#tests-requiring-optional-dependencies

@alexpearce - If you agree it's useful, maybe make that change in this PR?